tests/testthat/test_carp_clustering.R

```diff
@@ -63,3 +63,65 @@ test_that("Supplying percent and k gives error", {
 
   expect_error(clustering(carp_fit, k = 5, percent = 0.25))
 })
+
+
+test_that("CARP-returned sparsity pattern ends with full fusions",{
+  carp.fit <- CARP(presidential_speech)
+  expect_equal(
+    sum(carp.fit$carp.sol.path$v.zero.inds[,ncol(carp.fit$carp.sol.path$v.zero.inds)]),
+    nrow(carp.fit$carp.sol.path$v.zero.inds)
+  )
+})
+
+test_that("CARPVIZ-returned sparsity pattern begins with no fusions",{
+  carp.fit <- CARP(presidential_speech,alg.type = 'carpviz')
+  expect_equal(
+    sum(carp.fit$carp.sol.path$v.zero.inds[,1]),
+    0
+  )
+})
+test_that("CARPVIZ-returned sparsity pattern ends with full fusions",{
+  carp.fit <- CARP(presidential_speech,alg.type='carpviz')
+  expect_equal(
+    sum(carp.fit$carp.sol.path$v.zero.inds[,ncol(carp.fit$carp.sol.path$v.zero.inds)]),
+    nrow(carp.fit$carp.sol.path$v.zero.inds)
+  )
+})
+
+
+
+
+
+test_that("CARP-returned sparsity pattern begins with no fusions (uniform weights)",{
```

Comment:
Is this the right thing to check? What happens if, *e.g.*, we have two data points which are equal in all features (e.g., if GCleveland gets two rows instead of one)? Shouldn't they be fused at $\lambda = 0$ anyways?

Reply:
This is true. Perhaps we exclude these columns for testing in the case this happens? If so, I'll modify test to check for this.

Reply:
In the context of this test, it's no big deal. Just want to determine if this is something we're sensitive to generally.